documentation/projects/proposals/trust_and_safety/content_report_moderation/20231208-implementation_plan_django_admin_moderator_access.md

```diff
@@ -0,0 +1,491 @@
+Examples of potential future actions are:
+* `reversed_sensitive`, for reversing a sensitivity designation from a work previously
+  moderated as sensitive
+* `ignore_sensitive_text`, for marking a work to be ignored by sensitive text filtering
+* `reversed_deindex`, for reversing a deindex decision of a work
+* `submitted_metadata_correction`, for some future magical feature where we allow users
+  to submit suggested fixes to the metadata of a work
+
+Again, these are purely hypothetical. They're just meant as examples to drive home the
+_idea_ behind the action column. For now we will only implement the four actions listed
+in the columns list above.
+```
+
+#### Table view improvements
+
+We will scrap the existing report table. Instead of showing a list of reports,
+we'll show a list of reported media, sorted by the total number of pending
+reports, and then by the age of the oldest pending report. This will have the
+effect of sorting re-reported works to the top of the list, under the assumption
+that a work with multiple pending reports is probably higher priority, because a
+decision to report it was taken multiple times. By default, we will only show
+works with "pending" reports, with the option to include other statuses if
+moderators choose. This creates a queue to work through progressively, from top
+to bottom. Remove any manual sorting options for now. We can add those back if
+we get feedback that moderators want them, but for now they just clutter and
+confuse the interface, and make the foundational feature more complex.
+
+#### Media moderation view
+
+Clicking on single report should show the following:
+
+- The details of the reported media:
+  - title
+  - description
+  - tags
+  - the work itself
+  - whether sensitive text detection thought it had sensitive text
+  - provider/source
+  - links to foreign landing url and Openverse.org page for the work
+- A chronological list of all moderation decisions for the work
+- A chronological list of all reports for the work regardless of status
+
+The idea here is that the report view is not focused on taking an action for a
+single _report_, but rather showing all relevant information for a single work
+to make a moderation decision based on all pending reports for a work and
+historical moderation decisions for the work (if any).
+
+In the future, we could try to include links that open tables to show all
+moderation decisions for works by the same creator, etc, in case there is useful
+information, particularly for identifying potential bulk moderation actions.
+These kinds of additional views are out of scope for this current project,
+however.
+
+The expanded view should have the following context-sensitive actions:
+
+- "Mark sensitive: confirm a sensitive content report and mark the work as
+  sensitive"
+  - Only available if a work is not already marked sensitive
+  - `confirmed_sensitive` action
```

Comment:
Yep, there's nothing to prevent reports on media that's already marked sensitive. Keep in mind that reports aren't _only_ for sensitivity. The same report object handles DMCA. Additionally, something could be manually marked sensitive and get re-reported with a reason why it should be deindexed. In this case, the original moderation decision might have missed something that should have resulted in the work getting deindexed before. That's why it's "no_action" explicitly as "duplicate" if nothing happens on an already sensitive work, because there's nothing new to do.

However, now that I think about it, "duplicate" doesn't work if the report is DMCA. So maybe we shouldn't say "duplicate" but rather just "no_action" if the work is already marked sensitive? Maybe having two different "no action" statuses doesn't make sense right now and they could just both be covered under "rejected" meaning, the reports added no new information that led to a different moderation decision than whatever already stands?

Reply:
After reading [this other comment of yours](https://github.com/WordPress/openverse/pull/3494#discussion_r1423181916) I realised that a moderator could make any decision based on reports regardless of the stated report reason. There isn't a hard 1-1 relationship between marking a work sensitive and the report for the work having a "sensitive" reason. It could be "other" reason, for one, or even if something is reported as DMCA, it could not be a DMCA issue but also need to have the work marked sensitive.

I think the list of actions should be this then:

- `confirm_sensitive_keep` -> set all pending reports to `sensitive_filtered`
  - Only available if the work is not already marked sensitive
- `confirm_sensitive_deindex` -> set all pending reports to `deindexed` and deindex the work
- `confirm_dmca` -> set all pending reports to `deindexed` and deindex the work
- `dismiss` -> Anything that would result in "no action" regardless of context. Reducing this to a single action simplifies things a lot here with respect to how to handle the bulk update when rejecting sensitivity reports vs dmca reports. Moderators probably just want a way to say "none of the pending reports result in a change to the work's information" and be done with it. It would be annoying to have to reject sensitivity reports and dmca reports separately on a single work to fully clear it from the queue.

Having separate actions for sensitive and DMCA deindex actions lets us mark everything with the same single "deindexed" status while also recording the _reason_ why it was deindexed, and acknowledging that the reason a user chose to report a work is not always the reason the moderator made a particular choice. It also removes the complication of the report type also potentially being "other" which I wasn't handling at all before.

We'd have to get rid of this check, though, which filters the reason of the bulk-updated reports for decisions that didn't result in deindexing: https://github.com//WordPress/openverse/blob/aa16d4f1be7607b12c428886b9890bdd947cc71c/api/api/models/media.py#L238-L239

We could actually do well to remove the bulk-updating logic from the media report `save` and handle it in either `ModerationDecision`'s `save` method or, probably more reliably, in the admin view's method to save the form. It also helps separate the logic of report creation from report actioning instead of stuffing it into a single `save` method.

Reply:
+1 for the new list of actions, and separating out `confirm_sensitive_deindex` from `confim_dmca`!

I think it's still a bit unintuitive for me that taking an action updates the status of _all_ pending reports, when some of them might theoretically be unrelated/dismissable. Using the example where a record has several sensitive content reports that the moderator agrees with, and a dmca report that the moderator determines is invalid. They would take the action `confirm_sensitive_keep` and then all reports, including the invalid dmca report that wasn't really part of the decision, get set to `sensitive_filtered`. I guess that's not really an _error_, but it does feel like a slight misrepresentation of the decision.

> We could actually do well to remove the bulk-updating logic from the media report save and handle it in either ModerationDecision's save method or, probably more reliably, in the admin view's method to save the form. It also helps separate the logic of report creation from report actioning instead of stuffing it into a single save method.

Now that you say this, I think that would be excellent -- it makes much more sense to me that way.

Reply:
> It also removes the complication of the report type also potentially being "other" which I wasn't handling at all before.

Do we have any examples of what 'other' would be? Wondering if perhaps we need an additional `confirm_deindex` separate from `confirm_sensitive_deindex` and `confim_dmca` 🤔

Reply:
Here are a random 10 "other" reports from production:

```
+-----+--------------------------------------+--------+------------------------------------------------------------------------+----------------+-------------------------------+
| id  | identifier                           | reason | description                                                            | status         | created_at                    |
|-----+--------------------------------------+--------+------------------------------------------------------------------------+----------------+-------------------------------|
| 71  | e48fca1b-94da-46fa-a69c-90965ad5a619 | other  | Test description is 20 characters.                                     | pending_review | 2021-12-24 01:52:01.434245+00 |
| 388 | f9384235-b72e-4f1e-9b05-e1b116262a29 | other  | Testing the functionality of this field                                | no_action      | 2022-01-23 19:33:04.130104+00 |
| 407 | d7f3ed3f-0872-4012-9b63-dea62518c1fe | other  | the site is blocked and it wont let me use it                          | pending_review | 2022-01-27 16:12:56.817752+00 |
| 408 | 8d5a18ec-71d2-4161-9de2-3b1d9978076f | other  | what hapen two this im copi this en mi clas                            | pending_review | 2022-01-27 20:34:22.130701+00 |
| 411 | 2408d871-1c5f-40fc-8b2f-3421a1d45f88 | other  | License has been altered, is no longer Public Domain.                  | pending_review | 2022-02-02 19:41:57.068421+00 |
| 412 | f6ae6dde-701d-454e-9548-03df31cf8046 | other  | aint working                                                           | pending_review | 2022-02-02 19:50:33.699505+00 |
|     |                                      |        | stoping it                                                             |                |                               |
| 442 | 40df2568-a921-4252-bae9-110cf8b4d2ac | other  | There is no Kharkov Republic, this image supports separatist ideology. | pending_review | 2022-02-24 14:20:04.713105+00 |
| 463 | 832c00d2-cadb-49a0-b026-1875b71c3a09 | other  | inapropriot content kids use this                                      | pending_review | 2022-03-28 15:48:52.421191+00 |
| 479 | a2077e8f-ac72-462a-9309-4902f826bbe1 | other  | Wrong license in Open Verse                                            | pending_review | 2022-04-08 13:17:05.290978+00 |
| 480 | b0a52325-f249-4606-ad85-56cc7b85fec7 | other  | Wrong license in Open Verse                                            | pending_review | 2022-04-08 13:17:10.682073+00 |
+-----+--------------------------------------+--------+------------------------------------------------------------------------+----------------+-------------------------------+
```

Are there reasons we would deindex something that _aren't_ related to sensitivity or DMCA (or I suppose copyright issues more broadly)?

Some of these look like feedback. We have the wrong licence for some things. That's not DMCA or sensitivity, but we also have no way to action it :open_mouth:.

Maybe combining all report reasons into a single decision view is a terrible idea, actually, and we should just have separate queues for each report reason? But then have the same set of actions for all, considering the fact that any report could lead to any type of decision, even if just because the user chose the wrong type of report. The problem with other is that we can see that sometimes people use it when they mean DMCA or sensitivity (411 and 463 respectively).

This is really tricky! I'm starting to wonder if handling reports _separately_ is actually the simplest approach to this, and we should just handle them chronologically, with previous decisions visible in the moderation view, but not having _any_ bulk handling of reports?

Maybe we can start with that, handling single reports at a time, and then if in the future moderators give us feedback that they need bulk actioning of reports for a single media item, then we can implement it with a clearer idea of the precise use case? It would simplify things for now without accidentally creating difficult-to-undo patterns?

Reply:
I think single reports at a time would be fine for the number of reports we have now, and could be expanded upon in the future 👍 But the moderation view as described, where you can still see the context of _all_ the reports for a single image regardless of their report reason, still seems valuable to me even if we're only updating one report at a time. I think you'd still want the full context of reports and historic moderation decisions.

Ideally it would be nice if from the moderation view, you could select one **or more** pending reports and then take an action on just the selected reports. Is that doable? I guess you'd want to only allow taking action on pending reports, which adds a complication 🤔 We could have `pending_reports` be an actionable tabular inline list, and then separate (non-actionable) lists of previous moderation decisions and non-pending reports?

Reply:
That's true. So we'll show a list, have a checkbox next to each, and then the actions only apply to the selected ones? If there's only a single report, then it'll be automatically selected?

The separated lists sounds good too.

Reply:
Interesting conversation. The same doubts were coming to me as I read 😄  I think separate views for each report reason is a good idea, that way, the bulk actions should apply only to reports with the same reason because I _assume_ the people reviewing and making decisions on sensitive work would not be the same as the people confirming a DCMA infraction? It's an option, but I'm okay with starting with single reports moderation too.

Reply:
Yeah, we'll start there :+1: The "other" reason really throws this whole thing for a loop, and the fact that a user could choose the wrong reason as well (which appears to happen for "other" especially, in fact). I'll re-write this to simplify it along the lines of this discussion and we can deal with complexities once we have people actually using the tools who can give feedback about what needs to change :slightly_smiling_face:

Reply:
I've updated the plan to require the moderator to select a set of reports to action, and updated the list of actions to be more flexible. Nothing will operate on unselected reports and moderators can mark any type of report as having led to a sensitivity decision or copyright decision (those are the only two we have for now).